main.go

```diff
@@ -92,6 +94,45 @@ func splitChannel(channel string) (string, string, error) {
 	return prefix, version, nil
 }
 
+func doesSliceContain(slice []string, item string) bool {
+	for _, s := range slice {
+		if s == item {
+			return true
+		}
+	}
+	return false
+}
+
+// isValidVersion checks if the given version is not nil and >= minVersion
+func isValidVersion(v *semver.Version, minVersion *semver.Version) bool {
+	return v != nil && v.Compare(minVersion) >= 0
+}
+
+// processEdges process the cincinnati graph edges and updates AvailableUpgrades
+func processEdges(graph *CincinnatiGraph, minVersion *semver.Version, releases VersionReleases) {
+	for _, edge := range graph.Edges {
+		if len(edge) < 2 {
+			continue
+		}
+		fromIdx, toIdx := edge[0], edge[1]
+		if fromIdx < 0 || fromIdx >= len(graph.Nodes) || toIdx < 0 || toIdx >= len(graph.Nodes) {
+			continue
+		}
```

Comment:
shouldn't this return an error instead of ignoring it?

Reply:
done, PTAL.